### DISCUSSION TITLE
開発方針

### DISCUSSION THREAD

**Discussion by tomishima2904** [tomishima2904/shapley_association_game#15] [Ideas]
仕様や技術選定、アイデアなど開発に関すること

**Comment by tomishima2904** [comment]
ReactとDjangoの接続あたりで骨が折れそうなので、とりま素のHTMLとTailwindで作って、余力があればReactで書いてもいいかもしれん

**Comment by tomishima2904** [comment]
デザイン案は共有しているGoogle Driveフォルダ内にある`ワイヤーフレーム.drawio`で話し合おう
(Google DriveのリンクはDiscordに貼ってある)

**Comment by tomishima2904** [comment]
@ttoitt 語連想課題を解かせるときにユーザーにはどのような文言を提示すればよい？
"〇〇から連想される言葉は？" でいいの？ また、全問共通してこの文言で良い？

**Reply by ttoitt** [reply]
BERTに入力した文言は

###############################################
〇、〇、〇、〇、〇 は「MASK」の仲間です。
〇、〇、〇、〇、〇 からできているものは「MASK」です。
〇、〇、〇、〇、〇 から連想する色は「MASK」です。
〇、〇、〇、〇、〇 から連想する都道府県は「MASK」です。
〇、〇、〇、〇、〇 から連想する国は「MASK」です。
〇、〇、〇、〇、〇 から連想するスポーツは「MASK」です。
〇、〇、〇、〇、〇から連想する季節は「MASK」です。
〇、〇、〇、〇、〇から連想する場所は「MASK」です。
家の中で〇、〇、〇、〇、〇がある場所は「MASK」です。
〇、〇、〇、〇、〇から連想する行事は「MASK」です。
〇、〇、〇、〇、〇を使ってすることは「MASK」です。
〇、〇、〇、〇、〇は「MASK」のときに持っていきます。
〇、〇、〇、〇、〇から連想するメニューは「MASK」です。
###############################################

だから、これと同じ形でユーザにも解いてもらおうと思ってる。

**Reply by tomishima2904** [reply]
なるほどー
もし可能ならその入力文があるcsvとかtxtファイルがあるならくれると助かるかも！

**Reply by ttoitt** [reply]
[https://cdn.discordapp.com/attachments/1044561649559937075/1044582554403078144/keywords_fit2022.csv](url)
使う刺激語はDiscordに置いたやつ使う！

FITのときは、上の"〇"のところに刺激語突っ込む形でやってたから入力文にはなってない。
ちょっと待ってもらえれば入力文にした87題も作れる！
（今日は時間ないから明日になる、、、）

**Reply by tomishima2904** [reply]
刺激語のリストしかデータとしてはない感じか！おけい！
そしたら俺がFIT用に使ってたデータを利用して、連想ゲーム用で使うデータの雛形作るよ。
俺がFIT用で使ってたのが80題くらいで少し足りないくらいだから、問井君にはその雛形をもとに過不足を補う感じで頼もうかな。

**Reply by tomishima2904** [reply]
そしたら以下のような keyにカテゴリー番号(？)、valueにそのカテゴリー番号で使う質問文、を持つような`json`ファイル作ってくれ！
```
{0: ["{stim}", "は", "{mask}", "の", "仲間です。"], 1: ["{stim}", "からできているものは", "{mask}", "です。", ... }
```
以下の手順でgitを利用してやってみてくれい
1. まずは`develop/1`ブランチをローカルに作成して移動する
2. そこから`feature/1/q_sentences`というブランチを作成して移動する
3. `shapley_association_game/data`フォルダ内に`input_sentences.json`という名前で上記の`json`ファイルを作成
4.  作成できたら `git push origin feature/1/q_sentences`で push する
5. Githubの画面に行ってプルリクエストを作成する。この際、マージ先のブランチ名は`develop/1`であることに**注意**！

上記のことでわからないことがあったら、このスレッドでまたなんでも質問してくれい

(ちなみに`{stim}`は刺激語を表している)

**急いでないのでいつでも大丈夫！就活と研究をぜひ優先してくれ！**

**Reply by ttoitt** [reply]
ごめん。明日になりそう。

**お心遣いありがとう！助かる！！！**

**Reply by tomishima2904** [reply]
#38 に仕様（雛形）を書いたのでそれに即して書くと良い

**Comment by tomishima2904** [comment]
【画像に関して】
画像はJPEGやPNGではなくてベクトル形式になっている**SVG**形式にすると良い。
また、画像系はSVGファイルなら`static/svg`とフォルダを作成してそこにぶち込んでくれると助かる

アイコンの場合は`.ico`ファイルを作ると良い. コンバーターは[こちら](https://convertio.co/ja/svg-ico/)

**Reply by tomishima2904** [reply]
ちなみに画像のパスは`href="{% static 'svg/hoge.svg' %}"`のように書けば使える

**Reply by tomishima2904** [reply]
あと`base.html`に以下を追記すると[favicon](https://genchan.net/it/programming/python/django/12842/#ico)を設定できる
```
<link rel="icon"  href="{% static 'ico/favicon.ico' %}">
```

**Comment by tomishima2904** [comment]
【デザイン】
全部の画面をレスポンシブにしたい
[公式](https://tailwindcss.com/docs/responsive-design)によるとモバイルのレイアウトを決めてからweb版を決めた方がいいらしい笑

**Reply by tomishima2904** [reply]
あとは
- フォントデザイン
- 刺激語選択ボタンのデザイン（配置）
- 結果画面の表デザイン

を考えるって感じ。

**Reply by tomishima2904** [reply]
[TailwindCSSのレスポンシブデザイン](https://qiita.com/senju797/items/ad85794ec927f6ebd7bc)

**Reply by tomishima2904** [reply]
[Chromeでレスポンシブ対応を確認方法](https://beginners-hp.com/create_smartphone_check.html#:~:text=%E7%A2%BA%E8%AA%8D%E3%81%99%E3%82%8B%E3%83%9A%E3%83%BC%E3%82%B8%E3%82%92%E9%96%8B%E3%81%84,M%E3%82%92%E5%90%8C%E6%99%82%E3%81%AB%E6%8A%BC%E3%81%97%E3%81%BE%E3%81%99%E3%80%82)
